lib/entities/parse_address_from_domain.dart

```diff
@@ -24,17 +25,21 @@ Future<ParsedAddress> parseAddressFromDomain(
     final domainParts = formattedName.split('.');
     final name = domainParts.last;
 
+    final yatService = getIt.get<YatService>();
+
     if (domainParts.length <= 1 || domainParts.first.isEmpty || name.isEmpty) {
       try {
-        final addresses = await fetchYatAddress(domain, ticker);
-
-        if (addresses?.isEmpty ?? true) {
+        if (domain.hasOnlyEmojis) {
+          final addresses = await yatService.fetchYatAddress(domain, ticker);
+          if(addresses?.isEmpty ?? true){
+            return ParsedAddress(addresses: [domain], parseFrom: ParseFrom.yatRecord);
+          }
           return ParsedAddress(
-              addresses: [domain], parseFrom: ParseFrom.yatRecord);
+              addresses: addresses.map((e) => e.address).toList(),
+              name: domain,
+              parseFrom: ParseFrom.yatRecord);
         }
```

Comment:
Move that part above before `OpenaliasRecord.formatDomainName(domain);` calling. Let firstly decide is it emoji or domain and then call needed instructions.